### DISCUSSION TITLE
"pq: column p.paused_by does not exist" after upgrade from v7.4.1 to v7.6.0

### DISCUSSION THREAD

**Discussion by coffee-squirrel** [concourse/concourse#7746] [Help & Support] [answered]
We've just upgraded (using `concourse-bosh-deployment`) a sandbox environment from Concourse v7.4.1 to v7.6.0 (skipping [v7.5.0](https://github.com/concourse/concourse/releases/tag/v7.5.0) to let things settle per the release notes), and are now seeing the dreaded "experiencing turbulence" page.  We have a stable pipeline in place to generate the manifest, do the `bosh deploy`, etc., which completed successfully for this upgrade, so I don't believe there were any issues with that part.

Specifically, I see the `/api/v1/jobs` and `/api/v1/pipelines` endpoints are returning HTTP 500 (although others, e.g. `/api/v1/resources` and `/api/v1/teams`, are fine, I can log in, etc.).

Looking at `/var/vcap/sys/log/web/web.stdout.log`, I see messages (not the `j.` and `p.` variants) like:
```
{"timestamp":"2021-10-29T13:38:13.199619750Z","level":"error","source":"atc","message":"atc.list-all-pipelines.failed-to-get-all-visible-pipelines","data":{"error":"pq: column p.paused_by does not exist","session":"4989"}}
{"timestamp":"2021-10-29T13:38:13.202180818Z","level":"error","source":"atc","message":"atc.list-all-jobs.failed-to-get-all-visible-jobs","data":{"error":"pq: column j.paused_by does not exist","session":"4991"}}
{"timestamp":"2021-10-29T13:38:13.433912798Z","level":"error","source":"atc","message":"atc.list-all-pipelines.failed-to-get-all-visible-pipelines","data":{"error":"pq: column p.paused_by does not exist","session":"5001"}}
{"timestamp":"2021-10-29T13:38:13.436061651Z","level":"error","source":"atc","message":"atc.list-all-jobs.failed-to-get-all-visible-jobs","data":{"error":"pq: column j.paused_by does not exist","session":"5003"}}
{"timestamp":"2021-10-29T13:38:18.439648476Z","level":"error","source":"atc","message":"atc.list-all-pipelines.failed-to-get-all-visible-pipelines","data":{"error":"pq: column p.paused_by does not exist","session":"5013"}}
{"timestamp":"2021-10-29T13:38:18.459025911Z","level":"error","source":"atc","message":"atc.list-all-jobs.failed-to-get-all-visible-jobs","data":{"error":"pq: column j.paused_by does not exist","session":"5015"}}
{"timestamp":"2021-10-29T13:38:18.561999961Z","level":"error","source":"atc","message":"atc.list-all-pipelines.failed-to-get-all-visible-pipelines","data":{"error":"pq: column p.paused_by does not exist","session":"5025"}}
{"timestamp":"2021-10-29T13:38:18.565321897Z","level":"error","source":"atc","message":"atc.list-all-jobs.failed-to-get-all-visible-jobs","data":{"error":"pq: column j.paused_by does not exist","session":"5028"}}
{"timestamp":"2021-10-29T13:38:22.053735576Z","level":"error","source":"atc","message":"atc.scheduler.tick.component-failed","data":{"error":"find jobs to schedule: pq: column j.paused_by does not exist","session":"26.96"}}
{"timestamp":"2021-10-29T13:38:22.196083663Z","level":"error","source":"atc","message":"atc.reaper.tick.build-reaper.failed-to-get-pipelines","data":{"error":"pq: column p.paused_by does not exist","session":"28.96.2"}}
```

I've recreated the `web` instance (`bosh recreate web`), but the issue persists.

Looking at https://github.com/concourse/concourse/search?q=%22paused_by%22&type=code , it seems like there should've been a DB migration that happened but didn't (given the error messages).  183ce5e8298b79e61fcdbd450077ae5897d4c671 (#7273) indicates `paused_by` was introduced in `v7.5.0`.

```
Name       Release(s)                     Stemcell(s)                                    Team(s)
concourse  backup-and-restore-sdk/1.18.9  bosh-vsphere-esxi-ubuntu-bionic-go_agent/1.36  -
           bpm/1.1.15
           concourse/7.6.0
           os-conf/19
           postgres/43
```

As this is a sandbox environment we're not overly concerned, but it's definitely going to make us stick to v7.4.1 for a while longer in our Prod environment.

Would anyone happen to have a suggestion on how to proceed?

**Comment by taylorsilva** [comment, marked as answer]
I wonder if there's an issue with migrations, going from 7.4.1 to 7.5.0 and/or 7.6.0?
That column was added in this PR: https://github.com/concourse/concourse/pull/7273
which landed in 7.5.0. We'll need to investigate to see if going from 7.4.1 to 7.5.0 causes this issue as well.

**Reply by coffee-squirrel** [reply]
I've attached [concourse_sandbox_migrations.txt](https://github.com/concourse/concourse/files/7443166/concourse_sandbox_migrations.txt), containing the contents of the ATC `migrations` and `migrations_history` tables.

Relevant to this upgrade, I see the following migrations were done:
```
  version   |            tstamp             | direction | status | dirty
------------+-------------------------------+-----------+--------+-------
 1630515495 | 2021-10-29 12:35:23.064274+00 | up        | passed | f
 1630515496 | 2021-10-29 12:35:23.096779+00 | up        | passed | f
 1630515497 | 2021-10-29 12:35:23.106211+00 | up        | passed | f
 1630515498 | 2021-10-29 12:35:23.112022+00 | up        | passed | f
 1630515499 | 2021-10-29 12:35:23.115993+00 | up        | passed | f
```

These are all migrations whose commits were introduced in `v7.6.0`-- no `v7.5.0` migrations show up.

**Reply by coffee-squirrel** [reply]
@taylorsilva Should I go ahead and create an issue from this?  I don't think we're doing anything overly unique, so it seems like there might be a problem.

**Reply by taylorsilva** [reply]
We made a fix / workaround for this. Let me get the PR number

**Reply by taylorsilva** [reply]
I guess it doesn't notify in the discussion but I did make an issue from this discussion: https://github.com/concourse/concourse/issues/7753
Related PR is in the issue thread

**Reply by coffee-squirrel** [reply]
Great, thanks!

**Comment by schmurfy** [comment]
I just got this by trying to migrate from `7.4.2` to `7.6.0`, I rollbacked to `7.4.2`.
What is the proper way to upgrade ?

**Reply by clarafu** [reply]
The proper way to upgrade is rolling back to 7.4.0 from 7.4.2 and then upgrading to a later minor version from there. 7.5.1 and 7.5.2 have some improper migrations that can only be fixed by rolling back to 7.4.0.

**Comment by Berber31** [comment]
@coffee-squirrel 
Did you Rollbacked dB to 7.4.0 successfuly ?
Did it solved ?

**Reply by coffee-squirrel** [reply]
Yes, we downgraded a lower environment from v7.6.0 to v7.4.0 (v7.4.0 --> v7.4.1 --> v7.6.0 --> v7.4.0), and our Prod environment from v7.4.1 to v7.4.0 (v7.4.0 --> v7.4.1 --> v7.4.0), and things appear to be stable again.  v7.4.1 introduced a deadlock issue that made it impractical to stay on that version.

We followed https://concourse-ci.org/concourse-web.html#downgrading and downgraded both the DB and Concourse binary.
